### PR TITLE
docs: add documentation for methods used to mount an UFS path to an Alluxio path

### PR DESCRIPTION
**Pull request by abnerferreira0212**
# Pull Request Merge Checklist
           
Task backlog identifier related to this PR: 1542

---

## As a DEVELOPER, I confirm that this PR **doesn't contain**:
- [x] Code lines with syntax errors
- [x] Methods and variables names that aren't clear or doesn't mean anything
- [x] Code that is out of the standard (Example: **PEP-8** for Python code)
- [x] Files with **imports** errors or without any utility
- [x] **TODOs** left behind
- [x] Tests that are being skipped
- [x] Methods placed on incorrect files or folders (Example: methods that are local, but should be placed on /utils)
- [x] Functions, methods or API responses that are publicly returning sensitive information

## As a DEVELOPER, I confirm that I:
- [x] Followed the Simbiose's standard for branch and commit naming
- [x] Concluded the task and achieved the goals according to the task planning I defined before starting to code
- [ ] Added all necessary database changes to the [SQL Migrations](https://github.com/s1mbi0se/sql-migrations)
- [ ] Created tests for every new flow and method on this Pull Request
- [ ] Added the necessary DEBUG logs
- [x] Checked my code using the Simbiose's Corner Cases Guide
- [ ] Merged the **develop** branch to this PR/branch before sending the PR for review
- [ ] Solved all possible merge conflicts before sending the PR for review
- [ ] Corrected all the problems and code smells identified by the quality tools
- [ ] Manually executed the necessary tests locally to make sure my changes didn't break anything
- [ ] All unitary and integration tests ran by the CI system are passing
- [ ] Check the performance of my changes or additions and they didn't make the system slower (I profiled the code)
- [ ] Updated the necessary documentation (Example: OpenAPI, Docstrings, Metadata Database Modeling)
- [ ] Made the self-review of the code before sending the PR for review
- [ ] Sent this PR for a peer-review
- [x] Made all the corrections appointed during the peer-review
- [x] Confirmed that this PR merge will not cause other systems or dependencies to stop working (Example: ShannonDB Python Client and ShannonDB)
- [ ] Checked all GitHub Actions related to this PR that can block my merge

---  

## As the TEAM MANAGER, I confirm that I:
- [ ] Checked that the code was developed following the best practices for the language, with the best possible readability and using the less possible quantity of code lines and complexity. 
- [ ] Checked if the Pull Request and commit messages follow the Simbiose’s defined standard 
- [ ] Checked if there is no possible merge conflicts before reviewing the PR
- [ ] Checked if the purpose of the task was reached with this PR
- [ ] Checked that the code was developed following the best practices for the language, with the best possible readability and using the less possible quantity of code lines and complexity. 
- [ ] Checked if all related PRs were created when other components were affected by this task (e.g. database changes on [SQL Migrations](https://github.com/s1mbi0se/sql-migrations))
- [ ] Checked that were created unit, integration and chaos tests for every new flow and method on this Pull Request
- [ ] Checked that no sensible information (e.g. users security data, internal security data, etc) was exposed by this PR
- [ ] Checked that code coverage for this PR is higher than 80%.
- [ ] Checked that there are no problems and code smells identified by the quality and security tools
- [ ] Checked that all files were created on the proper folders according to the Simbiose’s defined standard
- [ ] Checked that all classes, methods and variables on this PR follow the Simbiose’s pattern for this programming language
- [ ] Checked that all unitary and integration tests ran by the CI system are passing
- [ ] Checked that there are no performance issues created by these changes or additions
- [ ] Checked that all documentation is up to date according to this change (Example: OpenAPI, Docstrings, Metadata Database Modeling)
- [ ] Checked that this PR merge will not cause other systems or dependencies to stop working (Example: ShannonDB Python Client and ShannonDB)
- [ ] Checked all GitHub Actions related to this PR that can block the merge

---  

## As the QUALITY RESPONSIBLE, I confirm that I:
- [x] Checked if the Pull Request and commit messages follow the Simbiose’s defined standard 
- [x] Checked if there is no possible merge conflicts before reviewing the PR
- [x] Checked that were created unit, integration and chaos tests for every new flow and method on this Pull Request
- [x] Checked that no sensible information (e.g. users security data, internal security data, etc) was exposed by this PR
- [x] Checked that code coverage for this PR is higher than 80%.
- [x] Checked that there are no problems and code smells identified by the quality and security tools
- [x] Checked that all files were created on the proper folders according to the Simbiose’s defined standard
- [x] Checked that all classes, methods and variables on this PR follow the Simbiose’s code standard for this programming language
- [x] Checked that all unitary and integration tests ran by the CI system are passing
- [x] Checked that there are no performance issues created by these changes or additions
- [ ] Checked that all documentation is up to date according to this change (Example: OpenAPI, Docstrings, Metadata Database Modeling)
- [ ] Checked that this PR merge will not cause other systems or dependencies to stop working (Example: ShannonDB Python Client and ShannonDB)
- [ ] Checked all GitHub Actions related to this PR that can block the merge

## Important links:
1. [Simbiose Quality Guidelines](https://github.com/s1mbi0se/company/wiki/Quality-Topics#table-of-contents)
2. [Metadata Database Modeling](https://docs.google.com/spreadsheets/d/1Y8_k5xV86kTjgfjbi-PGsJcubUKhWlT2x8ecpnMlyR0/edit?usp=sharing)
3. [Simbiose's Corner Cases Guide](https://www.simbioseventures.com/documentos-e-guias/procedimentos-e-checklists/corner-cases-guide)

### CODE REVIEW

File: core/common/src/main/java/alluxio/AbstractClient.java [hunk 1]
```diff
@@ -122,6 +122,16 @@ public AbstractClient(ClientContext context, InetSocketAddress address,
    */
   protected abstract ServiceType getRemoteServiceType();
 
+  /**
+   * Gets the remote service version.
+   * <p>
+   * Returns a long value representing the version for the remote service
+   * utilizing the {@link #mVersionService}.
```
Comment (caiopavanelli) [inline core/common/src/main/java/alluxio/AbstractClient.java:129]:
```suggestion
   * using the {@link #mVersionService}.
```

File: core/common/src/main/java/alluxio/AbstractClient.java [hunk 2]
```diff
@@ -122,6 +122,16 @@ public AbstractClient(ClientContext context, InetSocketAddress address,
...
+   *
+   * @return  the remote service version
+   * @throws  AlluxioStatusException  if an unforeseen exception occurs while
```
Comment (caiopavanelli) [inline core/common/src/main/java/alluxio/AbstractClient.java:132]: The expression "unforeseen  exception" is not clear. The exception is known, what actually throws it is unknown.

File: core/common/src/main/java/alluxio/retry/CountingRetry.java [hunk 3]
```diff
@@ -39,6 +39,16 @@ public int getAttemptCount() {
     return mAttemptCount;
   }
 
+  /**
+   * Checks if a new attempt can be made given the maximum number for retries in this policy.
```
Comment (caiopavanelli) [inline core/common/src/main/java/alluxio/retry/CountingRetry.java:43]: attempt to do what ?

File: core/common/src/main/java/alluxio/retry/ExponentialTimeBoundedRetry.java [hunk 4]
```diff
@@ -63,7 +63,15 @@ protected Duration computeNextWaitTime() {
   }
 
   /**
-   * @return a builder
+   * Creates and returns a new builder for this retry policy.
```
Comment (caiopavanelli) [inline core/common/src/main/java/alluxio/retry/ExponentialTimeBoundedRetry.java:66]: what "this" represents?

File: core/common/src/main/java/alluxio/retry/SleepingRetry.java [hunk 5]
```diff
@@ -35,6 +35,22 @@ public int getAttemptCount() {
     return mAttemptCount;
   }
 
+  /**
+   * Checks if a new attempt can be made. Sleeps between retries.
```
Comment (caiopavanelli) [inline core/common/src/main/java/alluxio/retry/SleepingRetry.java:39]: what is an attempt?

File: core/common/src/main/java/alluxio/retry/TimeBoundedRetry.java [hunk 6]
```diff
@@ -48,6 +48,15 @@ public int getAttemptCount() {
     return mAttemptCount;
   }
 
+  /**
+   * Checks if a new attempt can be made before a certain period of time has elapsed.
```
Comment (caiopavanelli) [inline core/common/src/main/java/alluxio/retry/TimeBoundedRetry.java:52]: what is an attempt ?

File: core/common/src/main/java/alluxio/security/authentication/ChannelAuthenticator.java [hunk 7]
```diff
@@ -75,9 +75,11 @@ public ChannelAuthenticator(GrpcConnection connection, Subject subject, AuthType
   }
 
   /**
-   * It builds an authenticated channel.
+   * Builds an authenticated channel.
    *
-   * @throws AlluxioStatusException
+   * @throws  AlluxioStatusException  if an unforeseen exception occurs
```
Comment (caiopavanelli) [inline core/common/src/main/java/alluxio/security/authentication/ChannelAuthenticator.java:80]: change "unforeseen exception"

File: core/common/src/main/java/alluxio/underfs/UnderFileSystemWithLogging.java [hunk 8]
```diff
@@ -1199,6 +1199,11 @@ public boolean isSeekable() {
     return mUnderFileSystem.isSeekable();
   }
 
+  /**
+   *
```
Comment (caiopavanelli) [inline core/common/src/main/java/alluxio/underfs/UnderFileSystemWithLogging.java:1203]: missing info

File: shell/src/main/java/alluxio/cli/fs/command/MountCommand.java [hunk 9]
```diff
@@ -81,6 +81,34 @@ public Options getOptions() {
         .addOption(OPTION_OPTION);
   }
 
+  /**
+   * Executes the {@code mount} command.
+   * <p>
+   * Keeps all command line arguments stored in a collection. Checks
+   * if this collection is empty, in which case information about the
+   * mount table with all actively synced paths is printed and the exit
```
Comment (caiopavanelli) [inline shell/src/main/java/alluxio/cli/fs/command/MountCommand.java:89]:
```suggestion
   * mount table with all actively synchronized paths is printed and the exit
```
